### PR TITLE
Shorter URLs + split "posts" into news+blog

### PR DESCRIPTION
**Pull request by ozkriff**
The gamedev.rs domain was recently transferred to me and I'm hoping to migrate https://rust-gamedev.github.io/ there soon (#233 _"Custom Domain"_).

In preparation to this, I also want to alter our URL scheme (mostly, because I want shorter URLs that would work nicely with Twitter, various chat apps, etc) while we're messing with the domain (and google cache) anyway.

The new scheme looks like "/\<short category name>/\<bare index (if possible)>":
- gamedev.rs/news/019
- gamedev.rs/blog/survey-01
- etc.

The new section structure is extensible: for example, I hope to integrate the https://rustgamedev.com podcast one day and URLs might look like gamedev.rs/cast/001

Live preview (for https://github.com/rust-gamedev/rust-gamedev.github.io/commit/0eaa2e9a5705b2a30df1b56ea5bc015540fee2f7): https://gamedev-rs.netlify.app

I've tried to add compatibility aliases for URLs (https://gamedev-rs.netlify.app/posts/newsletter-020 -> https://gamedev-rs.netlify.app/news/020, etc), hope I haven't missed anything important in this regard.

### CODE REVIEW

File: templates/includes/home.html [hunk 1]
```diff
@@ -26,16 +26,27 @@
...
+    {{ macros::post_list(posts=pages, placeholder="None yet - stay tuned for updates and announcements!") }}
+
+    <h2 class="post-list-heading">{{ section.title | default(value="Latest Blog Posts") }}</h2>
+
+    {% set posts = get_section(path="blog/_index.md") %}
     {% set pages = posts.pages | slice(end=config.extra.home.max_posts) %}
     {{ macros::post_list(posts=pages, placeholder="None yet - stay tuned for updates and announcements!") }}
 
     <p class="feed-subscribe">
         {% if pages | length > 0 %}
```
Comment (17cupsofcoffee) [inline templates/includes/home.html:40]: This 'pages' variable refers to the blog pages, not the news pages.

I think maybe let's just make these buttons display unconditionally? We're never going to go back to zero posts again :p

**Review by 17cupsofcoffee — APPROVED** [review]
This diff *looks* extremely intimidating, but if you filter out everything except .md/.html changes, it's not too scary 😛 

The new URLs look okay to me, and I've tried the redirects on your Netlify version and they all seem to work. One minor comment about the template, but that's it.

File: templates/includes/home.html (hunk 1 above)
Comment (ozkriff) [reply templates/includes/home.html]: Fixed

**Comment by ozkriff** [comment]
> This diff looks extremely intimidating, but if you filter out everything except .md/.html changes, it's not too scary stuck_out_tongue

Yeah, most of the changes are simple file renames

**Comment by ozkriff** [comment]
:timer_clock:  Merging tomorrow (and starting the actual migration to gamedev.rs) tomorrow if there're no objections

**Pull request merged**